### PR TITLE
Fix music images items in other languages

### PR DESCRIPTION
**Pull request by THEb0nny**
The problem has existed for a long time, icons for elements with music are not displayed.
If you look, then in Russian an undenfined error appears in the console.
```buttonImg.src = this.getSoundIcon(category);```

Here are some examples. English language.
image

That's it in Russian.
image
(These are pictures from the beta version.)

The problem was that if you enable another language on the site (for example, Russian), then the options values were taken with a translation in the selected language. They didn't match the category values (in English) in icons.jres.

1) Therefore, I decided to change it so that options sorting is disabled so that options do not change places in different languages.

2) Next, I from the initially obtained values
```soundIconCache = JSON.parse(pxtTargetBundle.bundledpkgs['music']['icons.jres']);```

image

I recycle the object so that all elements are with indices
```soundIconCacheArray = Object.entries(soundIconCache);```

I remove the first element from my array, it is not needed. Interferes.
```soundIconCacheArray.shift();```

3) In the refreshOptions function, I first get all the categories.
```const categories = this.getCategories(options);```
Then I look up the index in the array with all the categories of the category element
```const categoryIndex = categories.indexOf(category);```

Further, to set the picture for the musical element, I pass this index to getSoundIcon.
```buttonImg.src = this.getSoundIcon(categoryIndex);```

4) I rewrote the getSoundIcon function itself so that it takes an icon from my earlier soundIconCacheArray by index.

I checked in several languages. The changes are working.
As a result, we get this...
image

### CODE REVIEW

**Comment by THEb0nny** [comment]
I also made some improvements to the interface.

For example, I expanded them, made them always the same height, did work on the text.

It was like this before the change...
image
image
image

**Comment by THEb0nny** [comment]
@jwunderl But I have another question. I want to make it so that long words are wrapped with a hyphen.
I added style.hyphens = "auto" but it works if the correct language is in the html lang tag. It is now always displayed as lang="en". And hyphens doesn't work.
Another way is if the tag does not specify lang="ru". Therefore, I have a question, how to find out the language that is included on the site in order to write it into the code in the tag?

**Comment by THEb0nny** [comment]
@jwunderl please see changes

File: fieldeditors/field_music.ts [hunk 1]
```diff
@@ -202,15 +209,27 @@ export class FieldMusic extends pxtblockly.FieldImages implements Blockly.FieldC
...
+            buttonImg.src = this.getSoundIcon(categoryIndex);
+
             // Upon click/touch, we will be able to get the clicked element as e.target
             // Store a data attribute on all possible click targets so we can match it to the icon.
             const textNode = this.createTextNode_(content);
             button.setAttribute('data-value', value);
             buttonImg.setAttribute('data-value', value);
+            buttonImg.style.height = "auto";
             textNode.setAttribute('data-value', value);
+            //textNode.setAttribute('lang', "ru"); // for hyphens, here you need to set the correct abbreviation of the selected language
```
Comment (jwunderl) [inline fieldeditors/field_music.ts:226]:
```suggestion
            textNode.setAttribute('lang', pxt.Util.userLanguage()); // for hyphens, here you need to set the correct abbreviation of the selected language 
```

should work I believe?

File: fieldeditors/field_music.ts [hunk 2]
```diff
@@ -33,6 +34,8 @@ export class FieldMusic extends pxtblockly.FieldImages implements Blockly.FieldC
         }
         if (!soundIconCache) {
             soundIconCache = JSON.parse(pxtTargetBundle.bundledpkgs['music']['icons.jres']);
+            soundIconCacheArray = Object.entries(soundIconCache);
+            soundIconCacheArray.shift(); // Delete first elem
```
Comment (jwunderl) [inline fieldeditors/field_music.ts:38]:
```suggestion
            soundIconCacheArray = Object.entries(soundIconCache).filter(el => el[0] !== "*");
```

nit: I believe this should do the same thing, I prefer not to rely on json order being considered stable / etc

**Comment by THEb0nny** [comment]
@jwunderl indeed pxt.Util.userLanguage() works. I also found this in other files, but did not have time to try to apply it here.

**Comment by THEb0nny** [comment]
@jwunderl I checked, everything works. This change can be considered.

**Review by jwunderl — APPROVED** [review]
I had thought I merged / bumped this one, must have gotten distracted, sorry! Thanks for the pr!

**Pull request merged**